src/test/java/ReceiptAnalysisTest.java

```diff
@@ -214,4 +232,42 @@ public void analyzeImageAtThrowsIfTextRequestFails()
     Assert.assertEquals(TEXT_REQUEST_FAILED_WARNING, exception.getMessage());
     Assert.assertEquals(clientException, exception.getCause());
   }
+
+  private void stubAnnotationResponse(float confidenceScore) {
+    EntityAnnotation logoAnnotation =
+        EntityAnnotation.newBuilder().setDescription(STORE.get()).setScore(confidenceScore).build();
+    AnnotateImageResponse imageResponse =
+        createImageResponseWithText().addLogoAnnotations(logoAnnotation).build();
+    BatchAnnotateImagesResponse batchResponse =
+        BatchAnnotateImagesResponse.newBuilder().addResponses(imageResponse).build();
+    when(imageClient.batchAnnotateImages(anyList())).thenReturn(batchResponse);
+  }
+
+  private AnnotateImageResponse.Builder createImageResponseWithText() {
+    EntityAnnotation annotation = EntityAnnotation.newBuilder().setDescription(RAW_TEXT).build();
+    return AnnotateImageResponse.newBuilder().addTextAnnotations(annotation);
+  }
+
+  private void stubTextClassification() {
+    ClassificationCategory category =
+        ClassificationCategory.newBuilder().setName(CATEGORY_NAME).build();
+    ClassifyTextResponse classifyResponse =
+        ClassifyTextResponse.newBuilder().addCategories(category).build();
+    when(languageClient.classifyText(any(ClassifyTextRequest.class))).thenReturn(classifyResponse);
+  }
+
+  private ImmutableList<AnnotateImageRequest> createImageRequest() {
+    Image image = Image.newBuilder().setContent(IMAGE_BYTES).build();
+    List<Feature> features =
+        Arrays.asList(Feature.newBuilder().setType(Feature.Type.TEXT_DETECTION).build(),
```

Comment:
You could do ImmutableList.copyOf(Feature.newBuilder()...) to make this an ImmutableList instead of a List